### DISCUSSION TITLE
deploying a fresh site with one line module

### DISCUSSION THREAD

**Discussion by reggi** [freshframework/fresh#1141] [Ideas]
currently to my knowledge there's no way to deploy a fresh site as a third party module and then use that module to deploy a site. because fresh depends on the file system being intact at runtime, I'd love a way to be able to take a fresh site publish it as a third party module open up the deno deploy playground and create the server with a handful of lines, something like this

```ts
import site from "https://deno.land/x/reggi/site@1.0.0-1.6.3/index.ts";

await site({
    domain: Deno.env.get('DOMAIN')
})
```

I'm pretty sure this is isn't possible because the way that the islands routing works and looks at the local file system for the files in Islands folder, I think that this should be more dynamic in general and we should allow for remote Islands to exist as well this is another issue I think I've brought up in other discussions

this feature would potentially allow anybody to create a Shopify wrapper that can be used by everybody, and then you could pass in your Shopify API credentials and have a shop that's ready to go have a config field for the logo for example.

I think there's a lot of usage out of getting a portable server.

**Comment by marvinhagemeister** [comment]
I really like that idea and it's something we've been thinking about recently. Should be possible to create routes, islands and everything else without relying on the file system notation. Hoping to add that in the future.

**Reply by reggi** [reply]
Thanks!

While considering this idea, you may also consider allowing for "remote" islands. https://github.com/denoland/fresh/discussions/693 which would allow userland to create fresh-first interactive component-libraries expanding on the static stuff like https://fresh.deno.dev/components (which to my knowledge need to be static / non-interactive) or at least a file needs to be created in the islands directory as a stub :(

**Reply by marvinhagemeister** [reply]
Remote islands is something I've been thinking about as well, but I don't have a clear action plan yet. Something like
```js
import Foo from "bar" assert { fresh: "island" }
```

would be kinda cool and I think would fit into deno's spirit. But would require a bunch of tooling work maybe even in deno.

**Reply by reggi** [reply]
Still struggling with this almost a year later. There's this great [fediverse blog example here](https://github.com/dahlia/fedify/tree/main/examples/blog) but it uses fresh, and in order to use this I have to create a repo and copy the code and every time I want one of these blogs, then syncing all the repos is a pain, where in reality It could be a one file server that runs everything as a module hosted on `jsr`.

There's not much integration with fresh but decoupling it is a pain. Ideally a little bit of modularity and composability, like being able to make a fresh server a module, and include it in another fresh project would be nice, or just the ability to package up a fresh app to have one export would be nice. Dependence on the file system similar to next.js makes this impossible.

**Reply by marvinhagemeister** [reply]
Remote islands are already possible in the upcoming Fresh 2.0 release which will be released soon. It also allows you to use islands directly from JSR.
